src/corelibs/U2Algorithm/src/misc/RollingMatrix.h

```diff
@@ -30,11 +30,10 @@ namespace U2 {
 
 class U2ALGORITHM_EXPORT RollingMatrix {
 public:
+    // Always check memory overflow on the client side.
     RollingMatrix(int _sizeX, int _sizeY)
         : sizeX(_sizeX), sizeY(_sizeY), column0(0) {
-        // TODO: check memory overflow on the client side!
         U2_ASSERT(sizeX >= 0 && sizeY >= 0);
-        U2_ASSERT(getMatrixSizeInBytes(sizeX, sizeY) < INT_MAX);
```

Comment:
How this change is related to the problem? If you want to fix the problem in the 'TODO' please to what is written in the TODO: fix the caller side (find algorithm).